src/it/java/uk/gov/companieshouse/company_appointments/tests/CompanyAppointmentControllerITest.java

```diff
@@ -3,177 +3,321 @@
+                .andExpect(MockMvcResultMatchers.status().isBadRequest());
+    }
+
+    @Test
+    @DisplayName("Patch endpoint returns 403 forbidden when internal app privileges is missing")
+    void testPatchNewAppointmentCompanyNameStatusAppPrivilegesMissing() throws Exception {
+        mockMvc.perform(patch("/company/{company_number}/appointments/{appointment_id}", COMPANY_NUMBER, APPOINTMENT_ID)
+                .contentType(MediaType.APPLICATION_JSON)
+                .header(X_REQUEST_ID, "5342342")
+                .header(ERIC_IDENTITY, "SOME_IDENTITY")
+                .header(ERIC_IDENTITY_TYPE, "key")
+                .header(ERIC_AUTHORISED_KEY_PRIVILEGES, "invalid"))
+                .andExpect(MockMvcResultMatchers.status().isForbidden());
+    }
+
+    @Test
+    @DisplayName("Patch endpoint returns 404 not found when appointment does not exist")
+    void testPatchNewAppointmentCompanyNameStatusAppointmentDoesNotExist() throws Exception {
+        PatchAppointmentNameStatusApi requestBody = new PatchAppointmentNameStatusApi()
+                .companyName("company name")
+                .companyStatus("active");
+
+        mockMvc.perform(patch("/company/{company_number}/appointments/{appointment_id}", COMPANY_NUMBER, "fakeid")
+                .contentType(MediaType.APPLICATION_JSON)
+                .header(X_REQUEST_ID, "5342342")
+                .header(ERIC_IDENTITY, "SOME_IDENTITY")
+                .header(ERIC_IDENTITY_TYPE, "key")
+                .header(ERIC_AUTHORISED_KEY_PRIVILEGES, "internal-app")
+                .content(objectMapper.writeValueAsString(requestBody)))
+                .andExpect(MockMvcResultMatchers.status().isNotFound());
+    }
+
+    @Test
+    @DisplayName("Patch endpoint returns 503 service unavailable when resource changed endpoint unavailable")
+    void testPatchNewAppointmentCompanyNameStatusApiServiceUnavailable() throws Exception {
+
+        when(resourceChangedApiService.invokeChsKafkaApi(any())).thenThrow(ServiceUnavailableException.class);
+
+        PatchAppointmentNameStatusApi requestBody = new PatchAppointmentNameStatusApi()
+                .companyName("company name")
+                .companyStatus("active");
+
+        mockMvc.perform(patch("/company/{company_number}/appointments/{appointment_id}", COMPANY_NUMBER, APPOINTMENT_ID)
+                .contentType(MediaType.APPLICATION_JSON)
+                .header(X_REQUEST_ID, "5342342")
+                .header(ERIC_IDENTITY, "SOME_IDENTITY")
+                .header(ERIC_IDENTITY_TYPE, "key")
+                .header(ERIC_AUTHORISED_KEY_PRIVILEGES, "internal-app")
+                .content(objectMapper.writeValueAsString(requestBody)))
+                .andExpect(MockMvcResultMatchers.status().isServiceUnavailable());
+    }
+
+    @Test
+    @DisplayName("Patch endpoint returns 503 service unavailable when resource changed endpoint cannot connect to CHS Kafka Api")
```

Comment:
This test name doesn't make sense as the resource changed endpoint exists within chs-kaka-api.

Reply:
Addressed in latest commit